apps/wallet-mobile/src/Receive/AddressDetail.tsx

```diff
@@ -1,25 +1,322 @@
-import React from 'react'
-import {StyleSheet, View} from 'react-native'
+import React, { useRef, useState } from 'react'
+import { defineMessages, useIntl } from 'react-intl';
+import { Alert, Dimensions, FlatList, Image, StyleSheet, TouchableOpacity, View } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
 import QRCode from 'react-native-qrcode-svg'
+import Share from 'react-native-share';
+import SkeletonPlaceholder from 'react-native-skeleton-placeholder'
+import ViewShot, { captureRef } from "react-native-view-shot";
 
-import {COLORS} from '../theme'
+import { CopyButton, Text } from '../components'
+import { COLORS, colors } from '../theme'
 
-type Props = {
+type ShareProps = {
+  address?: string
+  title?: string
+  addressDetails?: AddressDetailsProps
+}
+
+type AddressDetailsProps = {
   address: string
+  stakingHash?: string
+  spendingHash?: string
+  title?: string
+}
+
+export function AddressDetail({ address, title, addressDetails }: ShareProps) {
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const SCREEN_WIDTH = Dimensions.get('window').width;
+  const itemsPerPage = 1;
+
+  const data = [
+    { cardType: 'QRCode', title, address },
+    { cardType: 'Details', address, stakingHash: addressDetails?.stakingHash, spendingHash: addressDetails?.spendingHash, title }]
+
+  const totalPages = Math.ceil(data.length / itemsPerPage);
+  const circleIndex = Array.from({ length: totalPages }, (_, index) => index + 1);
+
+  const onPageChange = (event) => {
+    const offset = event.nativeEvent.contentOffset.x;
+    const index = Math.floor(offset / (itemsPerPage * SCREEN_WIDTH - 64));
+    setScrollPosition(index);
+  };
+
+  return (
+    <>
```

Comment:
If its the screen root, i needs `SafeAreaView` from `react-native-safe-area-context` with the proper edges.